Fix gauntlet diagnostics windows

Gyan-max opened the pull request:
<!-- START SECTION: any other pull request -->

This PR fixes an issue in `gauntlet` where diagnostic messages were being handled differently on Windows versus Unix-based systems, causing CI failures. I implemented a normalization function to ensure consistent handling of paths, line endings, and OS-specific error messages across platforms.

Before submitting this PR, please make sure:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added tests (when appropriate).
- [x] You have updated the README or other documentation to account for these changes (when appropriate).
- [x] You have added an entry to the relevant `CHANGELOG.md` (see ["keep a changelog"] for more information).
- [x] Your commit messages follow the [conventional commit] style.
- [x] Your PR title follows the [conventional commit] style.

<!-- END SECTION -->

[conventional commit]: https://www.conventionalcommits.org/en/v1.0.0/#summary
["keep a changelog"]: https://keepachangelog.com/en/1.1.0/

claymcleod commented:
Hey Gyan, you'll need to rebase your branch on the latest version of `main`. Here's a guide that talks about rebasing (https://git-scm.com/book/en/v2/Git-Branching-Rebasing) —you will eventually have to force push your branch to get it based properly.

peterhuene commented on gauntlet/src/lib.rs:
```diff
@@ -133,6 +133,19 @@ pub struct Args {
+
+    // Handle OS-specific error messages
+    s.replace(
+        "The system cannot find the file specified. (os error 2)",
```
The error in question is actually different from the one the file tests see.

On Windows, error code 2 is for "file not found" and error code 3 is for "path not found". Gauntlet is running into "path not found" errors because the import paths from `broad/warp` contain a directory that doesn't exist.

So I think we'll need this:

```suggestion
        "The system cannot find the path specified. (os error 3)",
```

peterhuene reviewed (commented):
Other than the one comment, I think this looks good.

Gyan-max commented:
@peterhuene @adthrasher I have added normalization for the windows `path not found` error (error code 3), which now will be mapped to the Unix "No such file or directory" message and also updated the tests to properly test both error types.

adthrasher commented on gauntlet/CHANGELOG.md:
```diff
@@ -7,44 +7,48 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
-#### Changed
+### Changed
 
 * Upgradted `wdl` crate dependencies
```
While your at it, can you fix this type? `Upgradted` -> `Upgraded`

Gyan-max replied:
I'm sorry for this silly mistake

peterhuene commented:
The test failure appears to be spurious and I haven't been able to reproduce it locally where the request is returning an empty response from the test URL. We have other URLs we use to the same test service which I haven't seen show up in CI logs as a failure, so I wonder if there's something odd about this particular endpoint.

At any rate, we need to run `cargo fmt` as the suggested commit caused a needed formatting change.

If you push it up, hopefully the test job will pass (I'll look into reducing hitting the failure).

a-frantz commented on gauntlet/CHANGELOG.md:
please revert the header changes. They were made as part of https://github.com/stjude-rust-labs/wdl/pull/387

adthrasher commented on gauntlet/CHANGELOG.md:
```diff
@@ -7,44 +7,48 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## Unreleased
 
-#### Added
+### Added
```
@a-frantz - Didn't you recently add the extra layer of indentation to this file?

peterhuene commented:
@claymcleod looks like this is blocked from merging due to the commit signing requirement (I'm ignoring the add labels job failure for now).

peterhuene commented:
@Gyan-max thanks for fixing this!

Gyan-max commented:
@peterhuene @adthrasher @a-frantz 
Thanks for the guidance and support throughout this fix, it’s always been a pleasure working with you guys Looking forward to contributing more.
